stan/math/prim/fun/one_hot_int_array.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef STAN_MATH_PRIM_FUN_ONE_HOT_INT_ARRAY_HPP
+#define STAN_MATH_PRIM_FUN_ONE_HOT_INT_ARRAY_HPP
+
+#include <stan/math/prim/err.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return an integer array with 1 in the k-th position and zero elsewhere
+ *
+ * @param K size of the array
+ * @param k position of the 1 (indexing from 1)
+ * @return An integer array of size K with all elements initialised to zero
+ * and a 1 in the k-th position.
+ * @throw std::domain_error if K is not positive, or if k is less than 1 or
+ * greater than K.
+ */
+inline std::vector<int> one_hot_int_array(int K, int k) {
```

Comment:
[optional]
Probably not worth refactoring, but a single implementation would work here.  I realize there'd need to be two fronting functions to work easily with the Stan language, but the core implementation is shared.  They'd probably need an enable-if to ensure the type is arithmetic.

That's true of most of these functions.

Reply:
I'll pass on this for now.